File: src/main/java/com/innercircle/surveryproject/modules/entity/SurveyItem.java
package com.innercircle.surveryproject.modules.entity;

import com.innercircle.surveryproject.modules.dto.SurveyItemDto;
import com.innercircle.surveryproject.modules.enums.ItemType;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

/**
 * 설문조사 항목 엔티티
 */
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class SurveyItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 이름
     */
    @Column(nullable = false)
    private String name;

    /**
     * 설명
     */
    @Column(nullable = false)
    private String description;

    /**
     * 설문조사 유형
     */
    @Enumerated(EnumType.STRING)
    private ItemType itemType;

    /**
     * 필수여부
     */
    private Boolean required;

    /**
     * 활성화여부
     */
    @Setter
    private Boolean active = Boolean.TRUE;

    @Setter
    @ManyToOne
    @JoinColumn(name = "survey_id")
    private Survey survey;

    /**
     * 항목 별 입력 내용
     */
    @ElementCollection
    private List<String> itemContentList;

    public SurveyItem(SurveyItemDto surveyItemDto) {
        this.name = surveyItemDto.getName();
        this.description = surveyItemDto.getDescription();
        this.itemType = surveyItemDto.getItemType();
        this.required = surveyItemDto.getRequired();
        this.itemContentList = surveyItemDto.getItemContentList();
    }

    public static SurveyItem from(SurveyItemDto surveyItemDto) {
        return new SurveyItem(surveyItemDto);
    }

    public static List<SurveyItemDto> convertToDto(List<SurveyItem> surveyItemList) {
        return surveyItemList.stream().map(SurveyItemDto::toDto).toList();
    }

    public static List<SurveyItemDto> convert(List<SurveyItem> surveyItemList) {
        return surveyItemList.stream().filter(SurveyItem::getActive).map(SurveyItemDto::toDto).toList();
    }

    public static List<SurveyItem> convertToEntity(@NotNull List<SurveyItemDto> surveyItemDtoList) {
        return surveyItemDtoList.stream().map(SurveyItem::from).toList();
    }

    public Long getSurveyId() {
        return this.survey.getId();
    }

    /**
     * dto를 기준으로 기존 값이랑 같은지 비교
     * @param surveyItemDto
     * @return
     */
    public boolean equalsDto(SurveyItemDto surveyItemDto) {
        return this.id.equals(surveyItemDto.getSurveyItemId())
            && this.name.equals(surveyItemDto.getName())
            && this.description.equals(surveyItemDto.getDescription())
            && this.itemType.equals(surveyItemDto.getItemType())
            && this.itemContentList.equals(surveyItemDto.getItemContentList());
    }

    /**
     * dto를 기준으로 설문조사 항목 업데이트
     * @param surveyItemDto
     */
    public void updateFromDto(SurveyItemDto surveyItemDto) {
        this.name = surveyItemDto.getName();
        this.description = surveyItemDto.getDescription();
        this.itemType = surveyItemDto.getItemType();
        this.itemContentList = surveyItemDto.getItemContentList();
    }

}

File: survey-data/src/main/kotlin/com/ic/surveydata/entity/SurveyItem.kt
package com.ic.surveydata.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import survey.type.ItemType
import java.time.LocalDateTime

@Entity
@Table(name = "survey_item")
class SurveyItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0

    @Column(name = "uuid", nullable = false, unique = true)
    val uuid: String = ""

    @Column(name = "form_uuid", nullable = false, unique = true)
    val formUuid: String = ""

    @Column(name = "name", nullable = false, unique = false)
    val name: String = ""

    @Column(name = "is_required", nullable = false, unique = false)
    val isRequired: Boolean = true

    @Column(name = "description", nullable = false, unique = true)
    val description: String = ""

    @Column(nullable = false, unique = true)
    val type: ItemType = ItemType.SHORT_ANSWER

    // TODO - 혹시 kotlinx.datetime 을 사용할 수 있는 방법을 찾아보자 ... !
    @Column(name = "created_at", nullable = false, unique = false)
    val createdAt: LocalDateTime = LocalDateTime.now()
}
